File: DecompiledGameScripts/cyberpunk/systems/inputContextSystem.swift

public class InputContextSystem extends ScriptableSystem {

  private let activeContext: inputContextType;

  private final func OnPlayerAttach(request: ref<PlayerAttachRequest>) -> Void {
    this.activeContext = inputContextType.RPG;
  }

  public final const func GetActiveContext() -> inputContextType {
    return this.activeContext;
  }

  public final const func IsActiveContextAction() -> Bool {
    return Equals(this.activeContext, inputContextType.Action);
  }

  public final const func IsActiveContextRPG() -> Bool {
    return Equals(this.activeContext, inputContextType.RPG);
  }

  private final func OnChangeActiveContextRequest(request: ref<ChangeActiveContextRequest>) -> Void {
    this.activeContext = request.newContext;
  }
}

File: Definitions/Game_Definitions/classes/InputContextSystem.lua
---@meta
---@diagnostic disable

---@class InputContextSystem : gameScriptableSystem
---@field activeContext inputContextType
InputContextSystem = {}

---@return InputContextSystem
function InputContextSystem.new() return end

---@param props table
---@return InputContextSystem
function InputContextSystem.new(props) return end

---@return inputContextType
function InputContextSystem:GetActiveContext() return end

---@return Bool
function InputContextSystem:IsActiveContextAction() return end

---@return Bool
function InputContextSystem:IsActiveContextRPG() return end

---@param request ChangeActiveContextRequest
function InputContextSystem:OnChangeActiveContextRequest(request) return end

---@param request gamePlayerAttachRequest
function InputContextSystem:OnPlayerAttach(request) return end

